pump/storage/storage.go

```diff
@@ -489,13 +490,24 @@ func (a *Append) resolve(startTS int64) bool {
 		return false
 	}
 
-	log.Warn("unknown commit stats", zap.Int64("start ts", startTS))
-
 	tikvQueryCount.Add(1.0)
 	primaryKey := pbinlog.GetPrewriteKey()
 	status, err := a.tiLockResolver.GetTxnStatus(uint64(pbinlog.StartTs), uint64(pbinlog.StartTs), primaryKey)
 	if err != nil {
-		log.Error("GetTxnStatus failed", zap.Int64("start ts", startTS), zap.Error(err))
+		log.Error("get commit status failed for unknown txn", zap.Int64("start ts", startTS), zap.Error(err))
+		return false
+	}
+
+	log.Info("got commit status for unknown txn",
+		zap.Int64("start ts", startTS),
+		zap.Uint64("commit ts", status.CommitTS()),
+		zap.Uint64("ttl", status.TTL()),
+		zap.Reflect("action", status.Action()),
+		zap.Bool("isDDL", pbinlog.GetDdlJobId() > 0))
+
+	// check TTL (whether the lock is valid)
+	if status.TTL() > 0 {
```

Comment:
We could use the logic like this so we won't miss extra path.
```
if if status.TTL() > 0 {
    // lock is still valid.
} else {
    // The transaction status is certain now.
    if if status.IsCommitted() {
    } else{
    }
}
```

Reply:
🤔 I don't know what's your `miss extra path` means?

Reply:
Just some usage format, it does not matter much.